### PR TITLE
Updated the DataConnector code by implementing Durable Function App.

### PR DESCRIPTION
**Pull request by jayeshprajapaticrest**
Change(s):
   - Updated existing data connector with the Durable Function

   Reason for Change(s):
   - Due to the client side timeout error

   Version Updated:
   - Updated to 3.1.0

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

**Comment by v-rbajaj** [comment]
Hi @jayeshprajapaticrest, thanks for raising this PR, we will review this PR by 27 Sep 2023.

**Comment by v-rbajaj** [comment]
Hi @jayeshprajapaticrest, thanks for raising this PR, we will review this PR by 27 Oct 2023.

**Comment by jayeshprajapaticrest** [comment]
> Hi @jayeshprajapaticrest, thanks for raising this PR, we will review this PR by 27 Oct 2023.

@v-rbajaj Sure, Will wait for response from you regarding the PR review status update.

**Comment by v-rbajaj** [comment]
Hi @jayeshprajapaticrest, sorry for the delay, will review the PR by 30 Oct 2023.

**Comment by v-rbajaj** [comment]
Hi @jayeshprajapaticrest, `Solutions\RubrikSecurityCloud\SolutionMetadata.json`  can you please add SolutionMetadata file here?

**Comment by v-rbajaj** [comment]
Hi @jayeshprajapaticrest, can you please add `Solutions\RubrikSecurityCloud\SolutionMetadata.json` file here?

After that can you please repackage this solution post it?

**Comment by jayeshprajapaticrest** [comment]
@v-rbajaj As this is the updated PR for the already merged solution of RubrikSecurityCloud and this file is already their. We just updated connector related files so it would not be the part of this PR changes.

**Comment by jayeshprajapaticrest** [comment]
@v-rbajaj Any update on this PR?
As this PR is for issue fix for the already available solution and this need fix needs to available publicly as early as possible. So I request you to expediate the process to get it merge.

**Comment by v-rbajaj** [comment]
Hi @jayeshprajapaticrest, we are trying to deploy maintemplate to check if the contents are working fine or not. But the "manage" button is missing.  We tried deploying other maintemplate and the manage button does appear fine, there is something wrong with the maintemplate. Please give us time to investigate on this issue and we will get back to you by 02 Nov 2023. 
image

**Comment by jayeshprajapaticrest** [comment]
> Hi @jayeshprajapaticrest, we are trying to deploy maintemplate to check if the contents are working fine or not. But the "manage" button is missing. We tried deploying other maintemplate and the manage button does appear fine, there is something wrong with the maintemplate. Please give us time to investigate on this issue and we will get back to you by 02 Nov 2023. image

@v-rbajaj 
No problem. Let us know if anything is required from our end.
Thanks!

File: Solutions/RubrikSecurityCloud/Data/Solution_RubrikSecurityCloud.json [hunk 1]
```diff
@@ -15,10 +15,10 @@
         "Playbooks/RubrikRansomwareDiscoveryAndVMRecovery/azuredeploy.json"
     ],
     "Data Connectors": [
-        "Data Connectors/RubrikWebhookEvents_API_FunctionApp.json"
+        "Data Connectors/RubrikDataConnector/Rubrik_FunctionApp.json"
```
Comment (v-rbajaj) [inline Solutions/RubrikSecurityCloud/Data/Solution_RubrikSecurityCloud.json:18]: Hi @jayeshprajapaticrest, we can't find this file on the given path. And so we are getting this error while trying to repackage it. Please update the correct update and repackage the solution. 

image

Comment (jayeshprajapaticrest) [reply Solutions/RubrikSecurityCloud/Data/Solution_RubrikSecurityCloud.json]: @v-rbajaj Updated the PR with the mentioned changes.

Comment (v-rbajaj) [reply Solutions/RubrikSecurityCloud/Data/Solution_RubrikSecurityCloud.json]: Hi @jayeshprajapaticrest, as mentioned earlier, the solution needs to be repackaged too.

Comment (jayeshprajapaticrest) [reply Solutions/RubrikSecurityCloud/Data/Solution_RubrikSecurityCloud.json]: @v-rbajaj I have make the folder path as mentioned in the above path and also used the same path in the SolutionMetadata.json file and re-packaged the solution. Please verify and review the PR.
Thanks

**Review by v-rbajaj — APPROVED** [review]

**Pull request merged**